File: 1572/solution.go
package _1572

func diagonalSum(mat [][]int) int {
	n := len(mat)
	res := 0
	for i := 0; i < n; i++ {
		res += mat[i][i] + mat[i][n - i - 1]
	}
	if n % 2 == 1 {
		res -= mat[n / 2][n / 2]
	}
	return res
}

File: 1572/Solution.java
class Solution {
    public int diagonalSum(int[][] mat) {
        int n = mat.length;
        int res = 0;
        for (int i = 0; i < n; i++) {
            res += mat[i][i] + mat[i][n - i - 1];
        }
        if (n % 2 == 1) {
            res -= mat[n / 2][n / 2];
        }
        return res;
    }
}